comp/core/flare/types/types.go

```diff
@@ -9,147 +9,27 @@
-	//
-	// The path for each file in 'srcDir' is passed to the 'shouldInclude' callback. If 'shouldInclude' returns true, the
-	// file is copies to the flare. If not, the file is ignored.
-	//
-	// 'destDir' is a path relative to the flare root (ex: "some/path/to/a/dir").
-	//
-	// The data of each copied file is NOT scrubbed of any sensitive informations before being copied. Only files
-	// already scrubbed should be added in this way (ex: agent logs that are scrubbed at creation).
-	//
-	// Example: CopyDir("/var/log/datadog/agent", "logs", <callback>) will copy files from "/var/log/datadog/agent/" to
-	// "logs/agent/" in the flare.
-	CopyDirToWithoutScrubbing(srcDir string, destDir string, shouldInclude func(string) bool) error
-
-	// PrepareFilePath returns the full path of a file in the flare.
-	//
-	// PrepareFilePath will create the necessary directories in the flare temporary dir so that file can be create, but will
-	// not create the file. This method should only be used when the data is generated by another program/library.
-	//
-	// Example: PrepareFilePath("db-monitoring/db-dump.log") will create the 'db-monitoring' directory at the root of the
-	// flare and return the full path to db-dump.log: "/path/to/the/flare/db-monitoring/db-dump.log".
-	PrepareFilePath(path string) (string, error)
-
-	// RegisterFilePerm add the current permissions for a file to the flare's permissions.log.
-	RegisterFilePerm(path string)
-
-	// RegisterDirPerm add the current permissions for all the files in a directory to the flare's permissions.log.
-	RegisterDirPerm(path string)
-
-	// Save archives all the data added to the flare, cleanup all the temporary directories and return the path to
-	// the archive file. Upon error the cleanup is still done.
-	// Error or not, once Save as been called the FlareBuilder is no longer capable of receiving new data. It is the caller
-	// responsibility to make sure of this.
-	//
-	// This method must not be used by flare callbacks and will be removed once all flare code has been migrated to
-	// components.
-	Save() (string, error)
-}
-
-// FlareCallback is a function that can be registered as a data provider for flares. This function, if registered, will
-// be called everytime a flare is created.
-type FlareCallback func(fb FlareBuilder) error
+// FlareCallback is called by the FlareBuilder to build the flare
+// see the aliased type for the full description
+type FlareCallback = flaredef.FlareCallback
 
 // Provider is provided by other components to register themselves to provide flare data.
 type Provider struct {
 	fx.Out
-
-	Provider FlareCallback `group:"flare"`
+	Callback flaredef.FlareCallback `group:"flare"`
 }
 
 // NewProvider returns a new Provider to be called when a flare is created
-func NewProvider(callback FlareCallback) Provider {
+func NewProvider(callback flaredef.FlareCallback) Provider {
 	return Provider{
-		Provider: callback,
+		Callback: callback,
```

Comment:
Why the rename?

Reply:
It was a cleanup I thought would be good to do here. My reasoning is that we mostly really care about the clients of this code creating their own FlareProvider, and not peaking inside to see the implementation. The field within this object is rarely interacted with (I think only tests access it) so renaming it isn't a big deal. Having a field called `Provider` inside an object called `FlareProvider` stutters, plus I take issue with using the same name for two things that are really quite different. `FlareProvider` is the type that provides the flare that clients create, while the `Provider` field is actually a literal callback function, so we should call it as such.

Reply:
Thank you for the clarification. 

The reasoning makes total sense, and as you mentioned the field is access on tests.